homeassistant/components/homekit_controller/climate.py

```diff
@@ -170,8 +182,57 @@ def get_characteristic_types(self) -> list[str]:
             CharacteristicsTypes.TEMPERATURE_COOLING_THRESHOLD,
             CharacteristicsTypes.TEMPERATURE_HEATING_THRESHOLD,
             CharacteristicsTypes.SWING_MODE,
+            CharacteristicsTypes.ROTATION_SPEED,
         ]
 
+    def _get_rotation_speed_range(self) -> tuple[float, float]:
+        return (
+            round(self.service[CharacteristicsTypes.ROTATION_SPEED].minValue or 0) + 1,
+            round(self.service[CharacteristicsTypes.ROTATION_SPEED].maxValue or 100),
+        )
+
+    @property
+    def fan_modes(self) -> list[str] | None:
+        """Return the available fan modes."""
+        if self.service.has(CharacteristicsTypes.ROTATION_SPEED):
+            return [FAN_OFF, FAN_LOW, FAN_MEDIUM, FAN_HIGH]
+        return None
```

Comment:
```suggestion
        return [FAN_OFF, FAN_LOW, FAN_MEDIUM, FAN_HIGH]
```

Yes, the `None` is unreachable because it will never be called unless the supported feature is set

```
        if self.service.has(CharacteristicsTypes.ROTATION_SPEED):
            features |= ClimateEntityFeature.FAN_MODE
```